lib/client-assets.php

```diff
@@ -1652,8 +1668,11 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 
 	$post_autosave = gutenberg_get_autosave_newer_than_post_save( $post );
 	if ( $post_autosave ) {
-		$editor_settings['autosave'] = array(
-			'editLink' => add_query_arg( 'gutenberg', true, get_edit_post_link( $post_autosave->ID ) ),
+		$editor_settings['autosave'] = array_merge(
```

Comment:
Looking more, I'm more inclined to think that anything which _can_ be fulfilled in the client by mere awareness of the edited post's IDs _should be_ (preloading fine to offset load delay), since it's not really a "setting" in the true sense of the word.

If it comes to it, I'm willing to relent to let this pass through in its current form though.

Reply:
The general approach would be to show the most we can given the data we know. If we need to know the availability of an autosave before a preview can be performed, we disable or otherwise prevent the user from being able to initiate that action. Or, in the very, very worst case (i.e. bad), we don't show the editor at all until all requisite data is known. This technically already occurs for having the post data to edit. We remedy the asynchronous nature of data availability by [preloading data](https://github.com/WordPress/gutenberg/blob/fba803e46ba930b09e3763d224e5abfdf0275ed2/lib/client-assets.php#L1426-L1434) such that a network request which would normally take place is preempted to defer to data made available by the server at load-time.

The indirectness of this solution is why I'd made a point to mention my willingness to relent 😄 But in general, the minimal necessary to edit a post (ideally its ID) is all we'd provide from the server, and the rest is issued as a dependency thereof within the client itself.

Reply:
Oh right! I understand now—of course it can be preloaded, sorry to make you mention that twice.

I've refactored this to use apiFetch and have preloaded autosave data. That does make this a much more minimal change which is nice. I wasn't sure if an effect was the right way to do this, but that's how it works at the moment.